cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -1714,25 +1714,29 @@ Slice::CsGenerator::writeOptionalSequenceMarshalUnmarshalCode(
                 }
                 else
                 {
-                    out << nl << "if (" << stream << ".readOptional(" << tag << ", " << getOptionalFormat(seq) << "))";
-                    out << sb;
-                    if (builtin->isVariableLength())
+                    if (isArray)
                     {
-                        out << nl << stream << ".skip(4);";
+                        out << nl << param << " = " << stream << ".read" << func << "Seq(" << tag << ");";
```

Comment:
We were not checking for this simple path where we provide helper methods (like `readLongSeq(tag)`) on InputStream. As a result, all the `readXxxSeq(tag)` methods were untested.